remotespark/livyclientlib/pandaspysparklivyclient.py

```diff
@@ -20,11 +24,34 @@ def no_records(self, records_text):
         return records_text == "[]"
 
     def get_columns_dataframe(self, columns_text):
-        records = list()
-        columns = eval(columns_text)
-        return pd.DataFrame.from_records(records, columns=columns)
+        return pd.DataFrame.from_records([], columns=self._extract_strings_from_array(columns_text))
+
 
     def get_data_dataframe(self, records_text):
-        json_data = eval(records_text)
-        json_array = "[{}]".format(",".join(json_data))
-        return pd.DataFrame(json.loads(json_array))
+        strings = self._extract_strings_from_array(records_text)
+        try:
+            return pd.DataFrame([json.loads(s) for s in strings])
+        except ValueError:
+            raise DataFrameParseException("Cannot parse object as JSON: '{}'".format(strings))
+
+
+    def _extract_strings_from_array(self, s):
+        """This method consumes a string of the form
+           ['abc...', 'def...', 'xyz...', ...]
+           and returns all of the strings embedded inside the array as a list of
+           strings. That is, given a string that represents a list of single-quoted
+           strings, parse the array and return the list of strings. This method can
+           only handle single-quoted strings (not double-quoted) and it must handle
+           any list of any length, including 0. Furthermore it must support escaping
+           single-quotes using a backslash. Ignores newlines and leading and
```

Comment:
It depends. I would like it this method and its contract stayed the same forever but it depends on what kind of output we can get from Livy.  If Livy varies in the way that it displays strings, then this method will fail and we'll have to adapt.